qiskit/_schema_validation.py

```diff
@@ -0,0 +1,249 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Validation module for validation against JSON schemas."""
+
+import json
+import os
+import logging
+import jsonschema
+
+from qiskit import QISKitError
+from qiskit import __path__ as qiskit_path
+
+logger = logging.getLogger(__name__)
+
+# _DEFAULT_SCHEMA_PATHS[name]: local_path
+_DEFAULT_SCHEMA_PATHS = {
+    'backend_configuration': 'schemas/backend_configuration_schema.json',
+    'backend_properties': 'schemas/backend_properties_schema.json',
+    'backend_status': 'schemas/backend_status_schema.json',
+    'default_pulse_configuration': 'schemas/default_pulse_configuration_schema.json',
+    'job_status': 'schemas/job_status_schema.json',
+    'qobj': 'schemas/qobj_schema.json',
+    'result': 'schemas/result_schema.json'}
+# Schema and Validator storage
+_SCHEMAS = {}
+_VALIDATORS = {}
+
+
+def _load_schema(file_path, name=None):
+    """Loads the QObj schema for use in future validations. Caches schema in
+       _SCHEMAS module attribute.
+       Args:
+            file_path(str): Path to schema.
+            name(str): Given name for schema. Defaults to file_path filename
+                       without schema.
+       Return:
+            schema(dict): Loaded schema.
+
+
+    """
+    if name is None:
+        # filename without extension
+        name = os.path.splitext(os.path.basename(file_path))[0]
+    if name not in _SCHEMAS:
+        with open(file_path, 'r') as schema_file:
+            _SCHEMAS[name] = json.load(schema_file)
+
+    return _SCHEMAS[name]
+
+
+def _create_validator(name, schema=None, check_schema=True,
+                      validator_class=None, **validator_kwargs):
+    """
+    Generate validator for JSON schema.
+
+    Args:
+        name (str): Name for validator. Will be validator key in
+                    `_VALIDATORS` dict.
+        schema (dict): JSON schema `dict`. If not provided searches for schema
+                       in `_SCHEMAS`.
+        check_schema (bool): Verify schema is valid.
+
+        validator_class (jsonschema.IValidator): JsonSchema IValidator
+                instance. Default behavior is to determine this from the schema
+                `$schema` field.
+
+        **validator_kwargs: Additional keyword arguments for validator.
+
+    Return:
+        jsonschema.IValidator: Validator for JSON schema.
+
+    Raises:
+        SchemaValidationError: Raised if validation fails.
+    """
+    if schema is None:
+        try:
+            schema = _SCHEMAS[name]
+        except KeyError:
+            raise SchemaValidationError('''Valid schema name or schema must
+                                           be provided.''')
+
+    if name not in _VALIDATORS:
+
+        if not isinstance(schema, dict):
```

Comment:
Non-blocking: I think this check is not our responsibility. Notice you allow class customization and it is possible the class would understand schemas in other formats rather than a `dict`. I would recommend using `try ... except` or simply allowing exceptions to bubble up. What you think is right.